### PR TITLE
use K3s maintained kubernetes

### PR DESCRIPTION
**Pull request by clyang82**
Use k3s maintained Kubernetes to reduce the memory consumption for the controlplan.

### Before:
```
NAME                                                    CPU(cores)   MEMORY(bytes)
multicluster-controlplane-76f89dc9ff-d4p55              23m          423Mi
```
### After:
```
NAME                                                    CPU(cores)   MEMORY(bytes)
multicluster-controlplane-86cc66f746-6z9c9              18m          329Mi
```

[What have you removed from upstream Kubernetes in k3s?]( https://github.com/k3s-io/k3s#what-have-you-removed-from-upstream-kubernetes)

This is a common point of confusion because it has changed over time. Early versions of K3s had much more removed than the current version. K3s currently removes two things:

- In-tree storage drivers
- In-tree cloud provider

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -211,30 +201,31 @@ exclude go.opentelemetry.io/otel/internal/metric v0.27.0
...
-	k8s.io/kubelet => k8s.io/kubelet v0.24.0
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.24.0
-	k8s.io/metrics => k8s.io/metrics v0.24.0
-	k8s.io/mount-utils => k8s.io/mount-utils v0.24.0
-	k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.24.0
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.24.0
+	k8s.io/api => github.com/k3s-io/kubernetes/staging/src/k8s.io/api v1.25.4-k3s1
+	k8s.io/apiextensions-apiserver => github.com/k3s-io/kubernetes/staging/src/k8s.io/apiextensions-apiserver v1.25.4-k3s1
+	k8s.io/apimachinery => github.com/k3s-io/kubernetes/staging/src/k8s.io/apimachinery v1.25.4-k3s1
+	k8s.io/apiserver => github.com/k3s-io/kubernetes/staging/src/k8s.io/apiserver v1.25.4-k3s1
```
Comment (morvencao) [inline go.mod:207]: We must replace all these packages? I thought we just need replace `k8s.io/apiserver`.

Comment (clyang82) [reply go.mod:207]: I saw their repo uses all packages in k3s-io. I guess they make changes except for `k8s.io/apiserver`.

**Comment by qiujian16** [comment]
I check k3s.io/kubernetes, it seems the code is almost same with the upstream. Do we know why it is less resource consumable than upstream?

**Comment by clyang82** [comment]
Yes. Maybe only this commit https://github.com/k3s-io/kubernetes/commit/3e63f76a71c6550ccc799f3709cb0244626718a7 affects us. But it does not make a big different. The memory reducing is contributed by upgrading Kubernetes to 1.25.4.

Let me close this PR in favor of https://github.com/open-cluster-management-io/multicluster-controlplane/pull/13

**Pull request closed without merging**